docs/site/tutorials/authentication/Authentication-Tutorial.md

```diff
@@ -0,0 +1,974 @@
+        content: {
+          'application/json': {
+            schema: UserProfileSchema,
+          },
+        },
+      },
+    },
+  })
+  @authenticate('jwt')
+  async printCurrentUser(
+    @inject(AuthenticationBindings.CURRENT_USER)
+    currentUserProfile: UserProfile,
+  ): Promise<UserProfile> {
+    return currentUserProfile;
+  }
+
+  // ...
+```
+
+The `/users/me` endpoint is decorated with
+
+```ts
+@authenticate('jwt')
+```
+
+and authentication will only succeed if a valid JWT token is provided in the
+`Authorization` header of the request.
+
+Basically, the
+[AuthenticateFn](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/providers/auth-action.provider.ts)
+action in the custom sequence `MyAuthenticationSequence` (discussed in a later
+section) asks
+[AuthenticationStrategyProvider](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/providers/auth-strategy.provider.ts)
+to resolve the registered authentication strategy with the name `'jwt'` (which
+is `JWTAuthenticationStrategy` and discussed in a later section). Then
+`AuthenticateFn` calls `JWTAuthenticationStrategy`'s `authenticate(request)`
+function to authenticate the request.
+
+If the provided JWT token is valid, then `JWTAuthenticationStrategy`'s
+`authenticate(request)` function returns the user profile. `AuthenticateFn` then
+places the user profile on the request context using the
+`AuthenticationBindings.CURRENT_USER` binding key. The user profile is available
+to the `printCurrentUser()` controller function in a variable
+`currentUserProfile: UserProfile` through dependency injection via the same
+`AuthenticationBindings.CURRENT_USER` binding key. The user profile is returned
+in the response.
+
+If the JWT token is missing/expired/invalid, then `JWTAuthenticationStrategy`'s
+`authenticate(request)` function fails and an
+[HTTP 401 UnAuthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401)
+is thrown.
+
+If an **unknown** authentication strategy **name** is specified in the
+`@authenticate` decorator:
+
+```ts
+@authenticate('unknown')
+```
+
+then
+[AuthenticationStrategyProvider](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/providers/auth-strategy.provider.ts)'s
+`findAuthenticationStrategy(name: string)` function cannot find a registered
+authentication strategy by that name, and an
+[HTTP 401 UnAuthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401)
+is thrown.
+
+So, be sure to specify the correct authentication strategy name when decorating
+your endpoints with the `@authenticate` decorator.
+
+### Creating a Custom Sequence and Adding the Authentication Action
+
+In a LoopBack 4 application with REST API endpoints, each request passes through
+a stateless grouping of actions called a
+[Sequence](https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/sequence.ts).
```

Comment:
Question: why are you linking to the file instead of https://loopback.io/doc/en/lb4/Sequence.html?